geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TiledRasterRDD.scala

```diff
@@ -100,31 +100,25 @@ abstract class TiledRasterRDD[K: SpatialComponent: JsonFormat: ClassTag] extends
     crs: String,
     resampleMethod: ResampleMethod
   ): TiledRasterRDD[K] = {
-    val layoutDefinition = Right(LayoutDefinition(extent.toExtent, layout.toTileLayout))
+    val layoutDefinition = LayoutDefinition(extent.toExtent, layout.toTileLayout)
 
     reproject(layoutDefinition, TileRDD.getCRS(crs).get, getReprojectOptions(resampleMethod))
   }
 
   def reproject(
-    scheme: String,
-    tileSize: Int,
-    resolutionThreshold: Double,
+    layoutType: LayoutType,
     crs: String,
     resampleMethod: ResampleMethod
   ): TiledRasterRDD[K] = {
-    val _crs = TileRDD.getCRS(crs).get
+    val convertedCRS = TileRDD.getCRS(crs).get
+    val layoutDefinition =
+      layoutType.layoutDefinition(convertedCRS, rdd.metadata.extent, rdd.metadata.cellSize)
```

Comment:
No, that won't do the trick by itself. The reprojection could bow the extent where just reproejcting the envelope won't actually cover all the reprojected cells. I think we may have densified the envelope to reproject it in the past, but I'll have to double check that.